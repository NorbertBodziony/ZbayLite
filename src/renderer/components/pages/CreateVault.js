import React from 'react'
import { Redirect } from 'react-router'
import PropTypes from 'prop-types'

import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import Paper from '@material-ui/core/Paper'
import { withStyles } from '@material-ui/core/styles'

import PageWrapper from '../ui/PageWrapper'
import Snackbar from '../ui/Snackbar'
import VaultCreator from '../../containers/VaultCreator'

const styles = {
  gridRoot: {
    'min-height': '100vh'
  },
  paper: {
    padding: '20px'
  },
  welcome: {
    width: '260px',
    'margin-right': '20px'
  },
  vault: {
    width: '260px'
  },
  button: {
    backgroundColor: '#8d8d8d',
    color: '#fff',
    '&:hover': {
      backgroundColor: '#737373'
    }
  }
}

export const CreateVault = ({
  classes,
  inProgress,
  finished,
  inProgressMsg,
  error,
  onCloseSnackbar
}) => (
  <PageWrapper>
    <Snackbar
      variant='loading'
      message={inProgressMsg}
      open={inProgress}
      fullWidth
    />
    <Snackbar
      variant='error'
      message={error}
      open={error.length > 0}
      onClose={onCloseSnackbar}
    />
    { finished && <Redirect to='/home' />}
    <Grid container justify='center' alignItems='center' className={classes.gridRoot}>
      <Grid item>
        <Paper className={classes.paper}>
          <Grid container direction='row' justify='center' alignItems='center'>
            <Grid item className={classes.welcome}>
              <Typography variant='h4' gutterBottom>
                Hey there!
              </Typography>
              <Typography variant='body1' align='justify' gutterBottom>
                Zbay vault is not configured on this computer. In order to proceed you need to choose a password for your secure vault.
              </Typography>
            </Grid>
            <Grid item>
              <VaultCreator className={classes.vault} buttonStyles={classes.button} />
            </Grid>
          </Grid>
        </Paper>
      </Grid>
    </Grid>
  </PageWrapper>
)

CreateVault.propTypes = {
  classes: PropTypes.object.isRequired,
  inProgress: PropTypes.bool.isRequired,
  inProgressMsg: PropTypes.string.isRequired,
  error: PropTypes.string,
  finished: PropTypes.bool.isRequired,
  onCloseSnackbar: PropTypes.func.isRequired
}

CreateVault.defaultProps = {
  inProgress: false,
  inProgressMsg: '',
  error: '',
  finished: false
}

export default withStyles(styles)(CreateVault)
