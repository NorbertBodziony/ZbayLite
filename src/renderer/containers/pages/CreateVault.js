import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import CreateVault from '../../components/pages/CreateVault'
import vaultSelectors from '../../store/selectors/vault'
import vaultHandlers from '../../store/handlers/vault'

export const mapDispatchToProps = dispatch => bindActionCreators({
  onCloseSnackbar: vaultHandlers.actions.clearError
}, dispatch)

export const mapStateToProps = state => ({
  exists: vaultSelectors.exists(state),
  locked: vaultSelectors.locked(state),
  error: vaultSelectors.error(state),
  unlocking: vaultSelectors.unlocking(state),
  loading: vaultSelectors.creating(state)
})

export const CreateVaultWrapper = ({
  exists,
  locked,
  error,
  unlocking,
  loading,
  onCloseSnackbar
}) => (
  <CreateVault
    inProgress={loading || unlocking}
    inProgressMsg={loading ? 'loading' : 'creating'}
    finished={exists && !locked}
    error={error}
    onCloseSnackbar={onCloseSnackbar}
  />
)

export default connect(mapStateToProps, mapDispatchToProps)(CreateVaultWrapper)
