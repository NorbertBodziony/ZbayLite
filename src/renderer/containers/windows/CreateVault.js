import React, { useEffect, useState } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { useInterval } from '../hooks'
import CreateVault from '../../components/windows/CreateVault'
import Loading from '../../containers/windows/Loading'
import vaultSelectors from '../../store/selectors/vault'
import nodeSelectors from '../../store/selectors/node'
import vaultHandlers from '../../store/handlers/vault'
import nodeHandlers from '../../store/handlers/node'
import electronStore from '../../../shared/electronStore'

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      onCloseSnackbar: vaultHandlers.actions.clearError,
      getStatus: nodeHandlers.epics.getStatus,
      createVault: (password) => vaultHandlers.epics.createVault({ password })
    },
    dispatch
  )

export const mapStateToProps = state => ({
  exists: vaultSelectors.exists(state),
  locked: vaultSelectors.locked(state),
  error: vaultSelectors.error(state),
  unlocking: vaultSelectors.unlocking(state),
  loading: vaultSelectors.creating(state),
  nodeConnected: nodeSelectors.isConnected(state),
  bootstrapping: nodeSelectors.bootstrapping(state),
  bootstrappingMessage: nodeSelectors.bootstrappingMessage(state),
  creating: vaultSelectors.creating(state)
})

export const CreateVaultWrapper = ({
  exists,
  locked,
  error,
  unlocking,
  loading,
  onCloseSnackbar,
  bootstrapping,
  bootstrappingMessage,
  nodeConnected,
  getStatus,
  createVault,
  creating
}) => {
  const isNewUser = electronStore.get('isNewUser')
  useInterval(getStatus, 5000)
  const [done, setDone] = useState(null)
  const [password, storePass] = useState(null)
  const [passwordPosted, setPasswordPosted] = useState(null)
  useEffect(
    () => {
      if (exists && !locked) {
        setDone(true)
      }
    },
    [exists, locked]
  )
  useEffect(
    () => {
      if (unlocking === true) {
        setDone(false)
      }
    },
    [unlocking]
  )
  useEffect(
    () => {
      if (passwordPosted && !bootstrapping && nodeConnected) {
        createVault(password)
      }
    },
    [passwordPosted, bootstrapping, nodeConnected]
  )
  const isVaultCreationComplete = passwordPosted && !bootstrapping && nodeConnected
  const isDev = process.env.NODE_ENV === 'development'
  return (!isDev && (passwordPosted || isNewUser)) ? <Loading message={bootstrapping ? bootstrappingMessage : null} /> : (
    <CreateVault
      inProgress={loading || unlocking}
      inProgressMsg={loading ? 'loading' : 'creating'}
      finished={done}
      error={error}
      storePass={storePass}
      setPasswordPosted={setPasswordPosted}
      onCloseSnackbar={onCloseSnackbar}
      passwordPosted={passwordPosted}
      isVaultCreationComplete={isVaultCreationComplete}
    />
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateVaultWrapper)
