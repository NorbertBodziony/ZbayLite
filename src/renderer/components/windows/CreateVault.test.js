/* eslint import/first: 0 */
jest.mock('../../containers/VaultCreator', () => {
  const VaultCreator = () => (<div>VaultCreator</div>)
  return VaultCreator
})
import React from 'react'
import { shallow } from 'enzyme'

import { CreateVault } from './CreateVault'
import { mockClasses } from '../../../shared/testing/mocks'

describe('CreateVault', () => {
  it('renders component', () => {
    const result = shallow(
      <CreateVault
        classes={mockClasses}
        onCloseSnackbar={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })

  it('renders inProgress', () => {
    const result = shallow(
      <CreateVault
        classes={mockClasses}
        onCloseSnackbar={jest.fn()}
        inProgress
        inProgressMsg='creating'
      />
    )
    expect(result).toMatchSnapshot()
  })

  it('renders error', () => {
    const result = shallow(
      <CreateVault
        classes={mockClasses}
        onCloseSnackbar={jest.fn()}
        error='test error'
      />
    )
    expect(result).toMatchSnapshot()
  })
})
