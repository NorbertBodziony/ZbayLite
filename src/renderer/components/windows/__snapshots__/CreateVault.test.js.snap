// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateVault renders component 1`] = `
<Memo(WithStyles(WindowWrapper))>
  <Memo(WithStyles(Snackbar))
    fullWidth={true}
    message=""
    open={false}
    variant="loading"
  />
  <Memo(WithStyles(Snackbar))
    message=""
    onClose={[MockFunction]}
    open={false}
    variant="error"
  />
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="gridRoot-mock"
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Paper))
        className="paper-mock"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          direction="row"
          justify="center"
        >
          <WithStyles(ForwardRef(Grid))
            className="welcome-mock"
            item={true}
          >
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h4"
            >
              Hey there!
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              align="justify"
              gutterBottom={true}
              variant="body1"
            >
              Zbay vault is not configured on this computer. In order to proceed you need to choose a password for your secure vault.
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <VaultCreator
              buttonStyles="button-mock"
              className="vault-mock"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(WindowWrapper))>
`;

exports[`CreateVault renders error 1`] = `
<Memo(WithStyles(WindowWrapper))>
  <Memo(WithStyles(Snackbar))
    fullWidth={true}
    message=""
    open={false}
    variant="loading"
  />
  <Memo(WithStyles(Snackbar))
    message="test error"
    onClose={[MockFunction]}
    open={true}
    variant="error"
  />
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="gridRoot-mock"
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Paper))
        className="paper-mock"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          direction="row"
          justify="center"
        >
          <WithStyles(ForwardRef(Grid))
            className="welcome-mock"
            item={true}
          >
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h4"
            >
              Hey there!
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              align="justify"
              gutterBottom={true}
              variant="body1"
            >
              Zbay vault is not configured on this computer. In order to proceed you need to choose a password for your secure vault.
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <VaultCreator
              buttonStyles="button-mock"
              className="vault-mock"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(WindowWrapper))>
`;

exports[`CreateVault renders inProgress 1`] = `
<Memo(WithStyles(WindowWrapper))>
  <Memo(WithStyles(Snackbar))
    fullWidth={true}
    message="creating"
    open={true}
    variant="loading"
  />
  <Memo(WithStyles(Snackbar))
    message=""
    onClose={[MockFunction]}
    open={false}
    variant="error"
  />
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="gridRoot-mock"
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Paper))
        className="paper-mock"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          direction="row"
          justify="center"
        >
          <WithStyles(ForwardRef(Grid))
            className="welcome-mock"
            item={true}
          >
            <WithStyles(ForwardRef(Typography))
              gutterBottom={true}
              variant="h4"
            >
              Hey there!
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              align="justify"
              gutterBottom={true}
              variant="body1"
            >
              Zbay vault is not configured on this computer. In order to proceed you need to choose a password for your secure vault.
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <VaultCreator
              buttonStyles="button-mock"
              className="vault-mock"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(WindowWrapper))>
`;
